DequeueImpl.java

```diff
@@ -0,0 +1,99 @@
+import java.util.Arrays;
+import java.util.zip.DeflaterOutputStream;
+
+public class DequeueImpl<E extends Object>{
```

Comment:
```suggestion
public class DequeueImpl<E>{
```
любое обобщение идет от Object, поскольку любой класс его наследует